clients/unshaded/src/main/java/tachyon/client/file/FileSystemMasterClient.java

```diff
@@ -77,6 +83,144 @@ protected void afterConnect() throws IOException {
+   * @throws IOException if an I/O error occurs
+   * @throws TachyonException if a Tachyon error occurs
+   */
+  public synchronized long create(final String path, final CreateOptions options)
+      throws IOException, TachyonException {
+    return retryRPC(new RpcCallableThrowsTachyonTException<Long>() {
+      @Override
+      public Long call() throws TachyonTException, TException {
+        return mClient.create(path, options.toThrift());
+      }
+    });
+  }
+
+  /**
+   * Creates a new directory.
+   *
+   * @param path the directory path
+   * @param options method options
+   * @return the uri referencing the newly created directory
+   * @throws IOException if an I/O error occurs
+   * @throws TachyonException if a Tachyon error occurs
+   */
+  public synchronized TachyonURI createDirectory(final TachyonURI path,
+      final CreateDirectoryOptions options) throws IOException, TachyonException {
+    return retryRPC(new RpcCallableThrowsTachyonTException<TachyonURI>() {
+      @Override
+      public TachyonURI call() throws TachyonTException, TException {
+        mClient.mkdir(path.getPath(), options.toThrift());
+        // TODO(calvin): Look into changing the master side implementation
+        return path;
+      }
+    });
+  }
+
+  /**
+   * Creates a new file.
+   *
+   * @param path the file path
+   * @param options method options
+   * @return the uri referencing the newly created file
+   * @throws IOException if an I/O error occurs
+   * @throws TachyonException if a Tachyon error occurs
+   */
+  public synchronized TachyonURI createFile(final TachyonURI path, final CreateFileOptions options)
+      throws IOException, TachyonException {
+    return retryRPC(new RpcCallableThrowsTachyonTException<TachyonURI>() {
+      @Override
+      public TachyonURI call() throws TachyonTException, TException {
+        mClient.create(path.getPath(), options.toThrift());
+        // TODO(calvin): Look into changing the master side implementation
+        return path;
+      }
+    });
+  }
+
+  /**
+   * Marks a file as completed.
+   *
+   * @param fileId the file id
+   * @param options the method options
+   * @throws IOException if an I/O error occurs
+   * @throws TachyonException if a Tachyon error occurs
+   */
+  public synchronized void completeFile(final long fileId, final CompleteFileOptions options)
+      throws IOException, TachyonException {
+    retryRPC(new RpcCallableThrowsTachyonTException<Void>() {
+      @Override
+      public Void call() throws TachyonTException, TException {
+        mClient.completeFile(fileId, options.toThrift());
+        return null;
+      }
+    });
+  }
+
+  /**
+   * Marks a file as completed.
+   *
+   * @param path the file path
+   * @param options the method options
+   * @throws IOException if an I/O error occurs
+   * @throws TachyonException if a Tachyon error occurs
+   */
+  public synchronized void completeFile(final TachyonURI path, final CompleteFileOptions options)
```

Comment:
Temporary, this PR does not remove the other client so we cannot remove these without breaking things.